datasets/README.md

```diff
@@ -1 +1,85 @@
 # Flower Datasets
+[![GitHub license](https://img.shields.io/github/license/adap/flower)](https://github.com/adap/flower/blob/main/LICENSE)
+[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/adap/flower/blob/main/CONTRIBUTING.md)
+![Build](https://github.com/adap/flower/actions/workflows/framework.yml/badge.svg)
+![Downloads](https://pepy.tech/badge/flwr)
+[![Slack](https://img.shields.io/badge/Chat-Slack-red)](https://flower.dev/join-slack)
+
+Flower Datasets (`flwr-datasets`) is a library to quickly and easily create datasets for federated learning. It is created by the `Flower Labs` team that also created Flower: A Friendly Federated Learning Framework. 
+
+
+Flower Datasets library supports:
+* **downloading datasets** - choose the dataset from Hugging Face's `dataset`,
+* **partitioning datasets** - customize the partitioning scheme,
+* **creating centralized datasets** - leave parts of the dataset unpartitioned (e.g. for centralized evaluation).
+
+Thanks to using Hugging Face's `datasets` used under the hood, Flower Datasets integrates with the following popular formats/frameworks:
+* Hugging Face,
+* PyTorch, 
+* TensorFlow, 
+* Numpy, 
+* Pandas, 
+* Jax,
+* Arrow.
+
+Create **custom partitioning schemes** or choose from the **implemented partitioning schemes**:
+* i.i.d. partitioning `IidPartitioner(num_partitions)`
+* more to come in future releases.
+
+
+# Installation
+## With pip
+Flower Datasets can be installed from PyPi
+```bash
+pip install flwr-datasets
+```
+Install with an extension:
+* for image datasets:
+```bash
+pip install flwr-datasets[vision]
+```
+* for audio datasets:
+```bash
+pip install flwr-datasets[audio]
+```
+
+If you plan to change the type of the dataset to run the code with your ML famework make sure to have it installed too.
+
+# Usage
+The Flower Datasets exposes `FederatedDataset(dataset, partitioners)` abstraction to represent the dataset needed for federated learning/analytics.
+It has two powerful methods that let you handle the dataset preprocessing. They are `load_partition(idx, split)` and `load_full(split)`.
+
+Here's a quick example of how to partition the MNIST dataset:
+```
+from flwr_datasets import FederatedDataset
+
+# The train split of the mnist dataset will be partitioned into 100 partitions
+mnist_fds = FederatedDataset("mnist", partitioners={"train": 100}
+
+mnist_partition_0 = mnist_fds.load_partition(0, "train")
+
+centralized_data = mnist_fds.load_full("test")
+```
```

Comment:
```suggestion
# Usage

The Flower Datasets exposes `FederatedDataset(dataset, partitioners)` abstraction to represent the dataset needed for federated learning/analytics. It has two powerful methods that let you handle the dataset preprocessing. They are `load_partition(idx, split)` and `load_full(split)`.

Here's a quick example of how to partition the MNIST dataset:

```
from flwr_datasets import FederatedDataset

# The train split of the mnist dataset will be partitioned into 100 partitions
mnist_fds = FederatedDataset("mnist", partitioners={"train": 100}

mnist_partition_0 = mnist_fds.load_partition(0, "train")

centralized_data = mnist_fds.load_full("test")
```
```